autoload/go/guru.vim

```diff
@@ -483,6 +483,51 @@ function! go#guru#SameIds() abort
   call s:run_guru(args)
 endfunction
 
+function! go#guru#NextId() abort
+  let same_ids = getmatches()
+  if len(same_ids) < 2
+    return
+  endif
+
+  let curpos = getcurpos()
+  let current_idx = 0
+  for item in same_ids
+    let itempos = item['pos1']
```

Comment:
Aside from Fatih's comment, you need to make sure you're looping only over `sameIds` match groups. People can have more matches defined from their vimrc or other plugins (check the `group` attribute).

For example, for me it will break because I have [this in my vimrc](https://github.com/Carpetsmoker/dotfiles/blob/master/vim/softwrap.vim):

    :call matchadd('SoftWrap', '\%82v')
    :call matchadd('HardWrap', '\%122v')